src/buffer/buffer.rs

```diff
@@ -457,10 +462,39 @@ mod tests {
     }
 
     #[test]
-    fn debug() {
-        let mut buf = Buffer::empty(Rect::new(0, 0, 12, 2));
-        buf.set_string(0, 0, "Hello World!", Style::default());
-        buf.set_string(
+    fn debug_empty_buffer() {
+        let buffer = Buffer::empty(Rect::ZERO);
+        let result = format!("{buffer:?}");
+        println!("{result}");
+        let expected = "Buffer {
+    area: Rect { x: 0, y: 0, width: 0, height: 0 }
+}";
+        assert_eq!(result, expected);
```

Comment:
While refactoring the tests I used a formatting like the following. Maybe that's a bit more readable?
Using an additional macro (indoc) feels like overkill…

```suggestion
        let expected = "
Buffer {
    area: Rect { x: 0, y: 0, width: 0, height: 0 }
}";
        assert_eq!(result, expected.trim());
```

Reply:
I don't generally like seeing code in the left margin like this, which is why indoc works well (handling the indentation correctly. If you want to avoid indoc, perhaps use `\n` here instead (that's also what comes out of any test failure, so there's a convenience factor there).